code/modules/mob/living/brain/posibrain.dm

```diff
@@ -9,20 +9,24 @@ GLOBAL_LIST_EMPTY(on_station_posis)
 	supervisors = "your laws" //No AI yet as you are just a cube
 	show_in_prefs = FALSE //No reason to show in preferences
 
-/datum/job/cyborg/posibrain/after_spawn_silicon(mob/living/silicon/robot/R, mob/M)
+/datum/job/cyborg/equip(mob/living/carbon/human/H, visualsOnly = FALSE, announce = TRUE, latejoin = FALSE, datum/outfit/outfit_override = null, client/preference_source = null)
+
 	var/obj/item/mmi/posibrain/P = pick(GLOB.on_station_posis)
 
 	//Never show number of current posis
 	current_positions = 0
 
-	if(!P.activate(R)) //If we failed to activate a posi, kick them back to the lobby.
+	if(!P.activate(H)) //If we failed to activate a posi, kick them back to the lobby.
 		//Code taken from "send to lobby" admin panel
 		var/mob/dead/new_player/NP = new()
-		NP.ckey = M.ckey
+		NP.ckey = H.ckey
 
 		to_chat(NP, "<span class='warning'>Failed to Late Join as a Posibrain. Look higher in chat for the reason.</span>")
+		qdel(H)
+		return NP
```

Comment:
Have you checked to see the consequences of returning a new player in the equip chain? This could cause some seriously unintended behaviours since the job equip code isn't made to handle not being able to equip afaik

Reply:
I've tested it by disabling ghost roles, and it doesn't seem to impact things too badly (it does work properly). One side effect is that it leaves a "new player" atom (I think) on the spawn space, but I can't open the view variables tab to check what type it is. I can't interact with it at all.

Reply:
It *should* leave a new player atom in the spawn space - that's where players go while they're in the lobby, they are just invisible. 

Why is this `return NP` instead of just `return` or `return FALSE` though?

Reply:
I was just matching the return type, although it's a fair point, so I'll update it to that. That does make more sense after thinking about it for a bit.